ADR/propagate-pull-secrets.md

```diff
@@ -0,0 +1,52 @@
+# 0. Support deploying images from Private Quay.io repositories
+
+Date: 2023-03-16
+
+## Status
+
+Provisional
+
+## Context
+
+* As a user, I would expect StoneSoup to generate a private Quay.io image repository if my source code repository is  private.
+* As a user, I would expect StoneSoup to gracefully deploy images that were pushed to private Quay.io repositories in every relevant environment.
+
+
+## Decision
+
+### Overview
+
+1. This proposal builds on the capabilties of the https://github.com/redhat-appstudio/image-controller which sets up a Quay.io image repository per `Component` and 'downloads' a robot account token into the user's namespace for builds to be able to push to it.
+
+2. This proposal builds on the capabilities of SPI's `RemoteSecret` API. See https://github.com/redhat-appstudio/book/pull/70/files
+
+
+#### When a new `Component` is created
+
+1. A new Quay.io image repository and robot account token is made available in the namespace that would only be able to "pull" the `Component`'s image.
```

Comment:
Can you clarify which namespace here?

* The namespace that represents the user's appstudio control plane workspace?
* The namespace represented by the Environment's associated DeploymentTarget?

I think it is the first one - but want to double check. (Maybe update the language to be more specific?)

Reply:
The appstudio control plane namespace, yes (sure, I'll update)